dotnet/Cluster.cs

```diff
@@ -0,0 +1,432 @@
+        {
+        }
+
+        private static ComponentResourceOptions MakeResourceOptions(ComponentResourceOptions? options, Input<string>? id)
+        {
+            var defaultOptions = new ComponentResourceOptions
+            {
+                Version = Utilities.Version,
+            };
+            var merged = ComponentResourceOptions.Merge(defaultOptions, options);
+            // Override the ID if one was specified for consistency with other language SDKs.
+            merged.Id = id ?? merged.Id;
+            return merged;
+        }
+    }
+
+    public sealed class ClusterArgs : Pulumi.ResourceArgs
+    {
+        [Input("clusterSecurityGroupTags")]
+        private InputMap<string>? _clusterSecurityGroupTags;
+
+        /// <summary>
+        /// The tags to apply to the cluster security group.
+        /// </summary>
+        public InputMap<string> ClusterSecurityGroupTags
+        {
+            get => _clusterSecurityGroupTags ?? (_clusterSecurityGroupTags = new InputMap<string>());
+            set => _clusterSecurityGroupTags = value;
+        }
+
+        [Input("clusterTags")]
+        private InputMap<string>? _clusterTags;
+
+        /// <summary>
+        /// The tags to apply to the EKS cluster.
+        /// </summary>
+        public InputMap<string> ClusterTags
+        {
+            get => _clusterTags ?? (_clusterTags = new InputMap<string>());
+            set => _clusterTags = value;
+        }
+
+        /// <summary>
+        /// Indicates whether an IAM OIDC Provider is created for the EKS cluster.
+        /// 
+        /// The OIDC provider is used in the cluster in combination with k8s Service Account annotations to provide IAM roles at the k8s Pod level.
+        /// 
+        /// See for more details:
+        ///  - https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+        ///  - https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html
+        ///  - https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/
+        ///  - https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/aws/eks/#enabling-iam-roles-for-service-accounts
+        /// </summary>
+        [Input("createOidcProvider")]
+        public Input<bool>? CreateOidcProvider { get; set; }
+
+        /// <summary>
+        /// The number of worker nodes that should be running in the cluster. Defaults to 2.
+        /// </summary>
+        [Input("desiredCapacity")]
+        public Input<int>? DesiredCapacity { get; set; }
+
+        [Input("enabledClusterLogTypes")]
+        private InputList<string>? _enabledClusterLogTypes;
+
+        /// <summary>
+        /// Enable EKS control plane logging. This sends logs to cloudwatch. Possible list of values are: ["api", "audit", "authenticator", "controllerManager", "scheduler"]. By default it is off.
```

Comment:
I assume we'll want to make this a relaxed enum at some point.

Reply:
Yep, there are a few places where it'd be ideal to use enums. I have some TODOs in the code that generates the schema around this and just filed https://github.com/pulumi/pulumi-eks/issues/470 to track.